### PR TITLE
Hamburger menu wrap text when big fonts are used

### PR DESCRIPTION
**Pull request by githengi**
PR for Issue [https://github.com/OpenSRP/opensrp-client-reveal/issues/54](https://github.com/OpenSRP/opensrp-client-reveal/issues/54)

### CODE REVIEW

File: opensrp-reveal/src/main/res/values/dimens.xml [hunk 1]
```diff
@@ -17,6 +17,7 @@
     <dimen name="drawer_separator_margin">20dp</dimen>
     <dimen name="drawer_drawable_padding">10dp</dimen>
     <dimen name="dialog_min_width_major">90%</dimen>
+    <dimen name="operator_top_margin">50dp</dimen>
```
Comment (vincent-karuri) [inline opensrp-reveal/src/main/res/values/dimens.xml]: Don't you need a dimension for tablets as well?

Comment (githengi) [reply opensrp-reveal/src/main/res/values/dimens.xml]: This is minimum margin and the value is used for all devices. Constraint  vertical weight and bias are used to position the view in bigger screens or when normal or small text is used in the mobile devices(space is available).

File: opensrp-reveal/src/main/java/org/smartregister/reveal/view/ListTasksActivity.java [hunk 2]
```diff
@@ -221,9 +223,25 @@ public void onDrawerStateChanged(int newState) {//do nothing
 
         NavigationView navigationView = findViewById(R.id.nav_view);
         View headerView = navigationView.getHeaderView(0);
-        int screenHeightPixels = getResources().getDisplayMetrics().heightPixels
-                - getResources().getDimensionPixelSize(R.dimen.hamburger_margin);
-        headerView.setLayoutParams(new LayoutParams(LayoutParams.MATCH_PARENT, screenHeightPixels));
+        headerView.getViewTreeObserver().addOnGlobalLayoutListener(new ViewTreeObserver.OnGlobalLayoutListener() {
+            @Override
+            public void onGlobalLayout() {
```
[the remark this answers is not in the document]
Comment (githengi) [reply opensrp-reveal/src/main/java/org/smartregister/reveal/view/ListTasksActivity.java]: Done

File: opensrp-reveal/src/main/java/org/smartregister/reveal/view/ListTasksActivity.java [hunk 3]
```diff
@@ -223,9 +225,26 @@ public void onDrawerStateChanged(int newState) {//do nothing
...
+                if (getResources().getDisplayMetrics().heightPixels
+                        < headerView.getHeight() + getResources().getDimensionPixelSize(R.dimen.hamburger_margin)) {
+                    headerView.setLayoutParams(new LayoutParams(LayoutParams.MATCH_PARENT, LayoutParams.MATCH_PARENT));
+                    View operator = headerView.findViewById(R.id.operator_label);
+                    ConstraintLayout.LayoutParams params = (ConstraintLayout.LayoutParams) operator.getLayoutParams();
+                    params.height = ConstraintLayout.LayoutParams.WRAP_CONTENT;
+                    operator.setLayoutParams(params);
+                } else {//content of hamburger menu fits on screen; set menu height to screen height
+                    int screenHeightPixels = getResources().getDisplayMetrics().heightPixels
+                            - getResources().getDimensionPixelSize(R.dimen.hamburger_margin);
```
Comment (vincent-karuri) [inline opensrp-reveal/src/main/java/org/smartregister/reveal/view/ListTasksActivity.java]: define `getResources().getDisplayMetrics().heightPixels` and ` getResources().getDimensionPixelSize(R.dimen.hamburger_margin)` at the top of the function as they're used in both conditionals.

Comment (githengi) [reply opensrp-reveal/src/main/java/org/smartregister/reveal/view/ListTasksActivity.java]: done

**Pull request merged**